packages/opds-web-client/src/DataFetcher.ts

```diff
@@ -70,8 +70,15 @@ export default class DataFetcher {
             url: url,
             headers: response.headers
           });
-        })
-      }).catch(reject);
+        });
+      }).catch(error => {
+        reject({
+          status: error.status,
+          response: error.message,
+          url: url,
+          headers: error.headers
```

Comment:
I'm not sure the error will have all these properties - for example, https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful says it can return a `TypeError`, which would only have `message`. Weird that Typescript doesn't complain.

Reply:
Yeah, the error value itself is the message string so there's no object.